spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestBaseWithCatalog.java

```diff
@@ -59,18 +70,47 @@ protected static Object[][] parameters() {
   }
 
   @BeforeAll
-  public static void createWarehouse() throws IOException {
+  public static void setUpAll() throws Exception {
     TestBaseWithCatalog.warehouse = File.createTempFile("warehouse", null);
     assertThat(warehouse.delete()).isTrue();
+    startRESTServer();
   }
 
   @AfterAll
-  public static void dropWarehouse() throws IOException {
+  public static void tearDownAll() throws Exception {
     if (warehouse != null && warehouse.exists()) {
       Path warehousePath = new Path(warehouse.getAbsolutePath());
       FileSystem fs = warehousePath.getFileSystem(hiveConf);
       assertThat(fs.delete(warehousePath, true)).as("Failed to delete " + warehousePath).isTrue();
     }
+    stopRESTServer();
+  }
+
+  private static void startRESTServer() throws Exception {
+    restServer = new RESTCatalogServer();
+    // prevent using already-in-use port when testing
+    System.setProperty("rest.port", String.valueOf(MetaStoreUtils.findFreePort()));
+    System.setProperty(CatalogProperties.WAREHOUSE_LOCATION, warehouse.getAbsolutePath());
+    // In-memory sqlite database by default is private to the connection that created it.
+    // If more than 1 jdbc connection backed by in-memory sqlite is created behind one
+    // JdbcCatalog, then different jdbc connections could provide different views of table
+    // status even belonging to the same catalog. Reference:
+    // https://www.sqlite.org/inmemorydb.html
+    System.setProperty(CatalogProperties.CLIENT_POOL_SIZE, "1");
```

Comment:
There was this conversation getting lost (not sure why): https://github.com/apache/iceberg/pull/11093#discussion_r1791488648